data/avro-schema/src/main/resources/avro/net/corda/data/p2p/app/AuthenticatedMessageHeader.avsc

```diff
@@ -1,31 +1,37 @@
 {
   "type": "record",
   "name": "AuthenticatedMessageHeader",
+  "doc": "A message that will be delivered by the p2p layer over an end-to-end (authenticated and optionally encrypted) session.",
   "namespace": "net.corda.p2p.app",
   "fields": [
     {
+      "doc": "The destination identity for this message.",
       "name": "destination",
       "type": "net.corda.data.identity.HoldingIdentity"
     },
     {
+      "doc": "The source identity of this message.",
       "name": "source",
       "type": "net.corda.data.identity.HoldingIdentity"
     },
     {
+      "doc": "A (time-to-live) timestamp after which this message will be dropped from the p2p layer.",
```

Comment:
Probably worth adding the unit and describing what null means.

(By the way, can we use [`duration`](https://avro.apache.org/docs/1.11.0/spec.html#Duration)?)

Reply:
good point, I'll add the unit and semantics of null.

Wrt to duration, we decided to use a UTC timestamp here on purpose to avoid skew effects from different systems having to calculate the expiry time. Note: even the timestamp doesn't completely eliminate issues due to clock skew, but it makes the API a bit easier to reason about.

Reply:
So, maybe `timestamp-millis`?

Reply:
TTL is an industry term, so I think it's better (as long as we document unit and semantics)

Reply:
[discussed offline with Yiftach] apparently he meant the logical type, not changing the name.

Reply:
This might be relevant: https://issues.apache.org/jira/browse/AVRO-2123